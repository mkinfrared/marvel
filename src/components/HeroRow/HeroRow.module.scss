@import "styles/colors";

.HeroRow {
  border: 1px solid $secondary-main;
  display: flex;
  height: 250px;
  margin-bottom: 20px;

  @media (max-width: 640px) {
    align-items: center;
    flex-direction: column;
    height: 600px;
  }

  &:not(:last-child) {
    border-bottom-color: transparent;
  }

  .thumbnail {
    border: 1px solid $secondary-main;
    width: 250px;

    img,
    svg {
      height: 100%;
      object-fit: cover;
      object-position: 0;
      width: 100%;
    }

    @media (max-width: 640px) {
      height: 400px;
      max-width: 400px;
      width: 100%;

      img,
      svg {
        height: 100%;
        width: 100%;
      }
    }
  }

  .info {
    display: flex;
    flex: 1;
    flex-direction: column;
    padding: 20px;

    h4 {
      font-size: 24px;
      font-weight: 800;
    }

    p {
      line-height: 1.4;
      margin-top: 20px;
      text-align: justify;
    }
  }
}
