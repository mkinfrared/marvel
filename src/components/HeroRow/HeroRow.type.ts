export interface HeroRowProps {
  id: number;
  name: string;
  description: string;
  imageSrc?: string;
}
