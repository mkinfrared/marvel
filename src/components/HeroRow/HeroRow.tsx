import React from "react";
import { Link } from "react-router-dom";

import useWindowResize from "hooks/useWindowResize";
import { Pages } from "routes/Routes.type";
import Image from "ui/Image";

import css from "./HeroRow.module.scss";
import { HeroRowProps } from "./HeroRow.type";

const HeroRow = ({ description, id, imageSrc, name }: HeroRowProps) => {
  const page = useWindowResize();

  let loadWidth = 250;
  let loadHeight = 250;

  if (page.x < 641) {
    loadWidth = 380;

    loadHeight = 380;
  }

  return (
    <div className={css.HeroRow}>
      <div className={css.thumbnail}>
        <Image
          src={imageSrc || ""}
          loadWidth={loadWidth}
          loadHeight={loadHeight}
        />
      </div>
      <div className={css.info}>
        <Link to={Pages.HERO + id}>{name}</Link>
        <p>{description}</p>
      </div>
    </div>
  );
};

export default React.memo(HeroRow);
