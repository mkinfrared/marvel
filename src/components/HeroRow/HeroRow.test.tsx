import { render } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";

import HeroRow from "./HeroRow";

jest.mock("ui/Image", () => () => "<Image />");

describe("<HeroRow />", () => {
  const id = 42;
  const name = "marklar";
  const description = "foobar";
  const props = { id, name, description };

  const Component = (
    <MemoryRouter>
      <HeroRow {...props} />
    </MemoryRouter>
  );

  it("should be defined", () => {
    const { container } = render(Component);

    expect(container).toBeDefined();
  });

  it("match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });
});
