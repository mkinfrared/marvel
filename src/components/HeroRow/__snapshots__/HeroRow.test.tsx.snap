// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroRow /> match the snapshot 1`] = `
<div>
  <div
    class="HeroRow"
  >
    <div
      class="thumbnail"
    >
      &lt;Image /&gt;
    </div>
    <div
      class="info"
    >
      <a
        href="/hero/42"
      >
        marklar
      </a>
      <p>
        foobar
      </p>
    </div>
  </div>
</div>
`;
